docs/src/adoc/index.adoc

```diff
@@ -783,6 +783,60 @@ List<String> catalogNames = h.queryMetadata(DatabaseMetaData::getCatalogs)
 ----
 
 
+
+
+=== Usage in asynchronous applications
+
+Calling Jdbc and by extension Jdbi is inherently a blocking operation.
+In asynchronous applications
+(where results are returned as a link:https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html[CompletionStage^]
```

Comment:
we have a macro for that: `link:{jdkdocs}/javax/sql/DataSource.html[DataSource^]`